src/txHashBuilder.h

```diff
@@ -54,29 +59,48 @@ typedef enum {
 	TX_HASH_BUILDER_IN_AUX_DATA = 800,
 	TX_HASH_BUILDER_IN_VALIDITY_INTERVAL_START = 900,
 	TX_HASH_BUILDER_IN_MINT = 1000,
-	TX_HASH_BUILDER_IN_MINT_TOP_LEVEL_DATA = 1010,
-	TX_HASH_BUILDER_IN_MINT_ASSET_GROUP = 1011,
-	TX_HASH_BUILDER_IN_MINT_TOKEN = 1012,
 	TX_HASH_BUILDER_IN_SCRIPT_DATA_HASH = 1100,
 	TX_HASH_BUILDER_IN_COLLATERALS = 1200,
 	TX_HASH_BUILDER_IN_REQUIRED_SIGNERS = 1300,
 	TX_HASH_BUILDER_IN_NETWORK_ID = 1400,
-	TX_HASH_BUILDER_FINISHED = 1500,
+	TX_HASH_BUILDER_IN_COLLATERAL_RETURN = 1500,
+	TX_HASH_BUILDER_IN_TOTAL_COLLATERAL = 1600,
+	TX_HASH_BUILDER_IN_REFERENCE_INPUTS = 1700,
+	TX_HASH_BUILDER_FINISHED = 1800,
 } tx_hash_builder_state_t;
 
+typedef enum {
+
+	TX_OUTPUT_TOP_LEVEL_DATA = 10,
+	TX_OUTPUT_ASSET_GROUP = 11,
+	TX_OUTPUT_TOKEN = 12,
+	TX_OUTPUT_DATUM_OPTION = 20,
+	TX_OUTPUT_DATUM_OPTION_CHUNKS = 21,
+	TX_OUTPUT_SCRIPT_REFERENCE = 30,
+	TX_OUTPUT_SCRIPT_REFERENCE_CHUNKS = 31,
+
+} tx_hash_builder_output_substate_t;
+
 typedef struct {
 	uint16_t remainingInputs;
 	uint16_t remainingOutputs;
 	uint16_t remainingWithdrawals;
 	uint16_t remainingCertificates;
 	uint16_t remainingCollaterals;
 	uint16_t remainingRequiredSigners;
+	uint16_t remainingReferenceInputs;
+	uint16_t totalDatumSize;
+	uint16_t currentDatumSize;
+	uint16_t totalReferenceScriptSize;
+	uint16_t currentReferenceScriptSize;
```

Comment:
I'd move these elsewhere, they are not direct tx body elements (as the rest of this list). Perhaps the union below?